cpp/ql/test/library-tests/ir/ir/PrintAST.expected

```diff
@@ -10449,7 +10479,37 @@ ir.cpp:
 # 1308|                 expr: [VariableAccess] y
 # 1308|                     Type = [LongType] long
 # 1308|                     ValueCategory = prvalue(load)
-# 1308|           1: [VariableAccess] x
+# 1308|           1: [CStyleCast] (int)...
+# 1308|               Conversion = [IntegralConversion] integral conversion
+# 1308|               Type = [IntType] int
+# 1308|               ValueCategory = prvalue
+# 1308|             expr: [ParenthesisExpr] (...)
+# 1308|                 Type = [BoolType] bool
+# 1308|                 ValueCategory = prvalue
+# 1308|               expr: [LogicalOrExpr] ... || ...
```

Comment:
Is this the exact same expression object as child 0, or is it a copy? What happens if operand 0 might have side effects, as in `foo() ?: 7`?

Reply:
In this case, this is a separate entity in the db, but getting the same entity is possible, such as in your `foo()` example.

You can see an example of an expr being reused like that in `question_mark_colon/types.expected` that I added in this PR.